File: 1 basic/1002. 写出这个数 (20).cpp
#include<iostream>
#include<string>
using namespace std;

string sum(string num){
	int n=0,i;
	string tmp,result="";
	for(i=0;i<num.size();++i){
		n+=num[i]-'0';
	}
	
	while(n){
		switch(n%10){
		case 0: tmp="ling ";	break;
		case 1: tmp="yi ";	break;
		case 2: tmp="er ";	break;
		case 3: tmp="san ";	break;
		case 4: tmp="si ";	break;
		case 5: tmp="wu ";	break;
		case 6: tmp="liu ";	break;
		case 7: tmp="qi ";	break;
		case 8: tmp="ba ";	break;
		case 9: tmp="jiu ";	break;
		}
		result = tmp + result;
		n/=10;
	}
	
	
	return result.substr(0,result.size()-1);
}

int main(){
	string num;
	cin>>num;
	cout<<sum(num)<<endl;
	
	return 0;
}

File: 1 basic/py/1002. 写出这个数 (20).py
from _functools import reduce

n = input()
def char2int(ch):
    return {'0':0,'1':1,'2':2,'3':3,'4':4,'5':5,'6':6,'7':7,'8':8,'9':9}[ch]
s = reduce(lambda x,y: x+y, map(char2int, n))
s = str(s)
def switch(ch):
    return {'0':'ling ','1':'yi ','2':'er ', 
            '3':'san ', '4':'si ', '5':'wu ',
            '6':'liu ', '7':'qi ', '8':'ba ','9':'jiu '}[ch]
ans = ''
for ch in s: ans = ans + switch(ch)
print(ans.strip())